### PR TITLE
More updates on rosdep rules guidelines

### PR DESCRIPTION
**Pull request by tfoote**
From thoughts over lunch and thinking back on past reviews.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,9 +39,20 @@ Guidelines for rosdep rules
 ---------------------------
 
  * Native packages are strongly preferred (They are required for packaging and have upgrade and conflict tracking.)
- * Python packages should go in the python.yaml
- * Homebrew into homebrew.yaml
+ * Please use the smallest notation possible and if something is default or not necessary to specify, don't.
```
Comment (wjwwood) [inline CONTRIBUTING.md]: I don't think what mean is clear here. The terms `notation` and `is default` aren't well defined imo. Maybe you should include examples of each.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -39,9 +39,20 @@ Guidelines for rosdep rules
...
+ * Python packages should go in the python.yaml.
```
Comment (wjwwood) [inline CONTRIBUTING.md]: nitpick: The use of `.` is very inconsistent, you added it here, but on the next line (which you also updated) it doesn't have it. Use it or don't, but being consistent is ideal.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -39,9 +39,20 @@ Guidelines for rosdep rules
...
+ * Homebrew into osx-homebrew.yaml
  * If a package is only available via pip use the `-pip` extension.
+ * Supported Platforms
+   * Rules can be contributed for any platform. However to be released they must be at least cover the supported platforms in REP 3: http://www.ros.org/reps/rep-0003.html So:
+    * Ubuntu and Debian are top priority.
+    * Fedora and Gentoo have packaging jobs should be best effort.
```
Comment (wjwwood) [inline CONTRIBUTING.md]: This isn't a sentence, maybe you started to say something and changed or there is a word missing?

**Comment by dirk-thomas** [comment]
@tfoote ping

**Comment by tfoote** [comment]
Comments addressed and rebased

**Review by wjwwood — APPROVED** [review]

**Pull request merged**